x509/pkix/pkix.go

```diff
@@ -54,6 +54,7 @@ type Name struct {
 	StreetAddress, PostalCode, DomainComponent []string
 	EmailAddress                               []string
 	SerialNumber, CommonName                   string
+	SerialNumbers, CommonNames                 []string
```

Comment:
Does there need to be an accompanying diff to `x509/json.go`'s `MarshalJSON` function and `JSONCertificate` type to handle these fields?

Reply:
I thought about this too, but I don't think that we necessarily want to output them in JSON. All but a handful of messed up certificates are not going to have more than one CN or Serial Number. I primarily wanted to expose so that we could write a lint that checks for this. If we indeed think that we need to output all in the JSON, I think that might be the time to switch CommonName and SerialNumber to be lists themselves instead of a singular value.

Reply:
:+1: Makes sense to me. The `JSONCertificate` godoc explicitly says "This struct has a distinct and independent layout from x509.Certificate, mostly for condensing data across repetitive fields ..." and this does seem like repetitive data.